include/wx/string.h

```diff
@@ -1824,7 +1824,11 @@ class WXDLLIMPEXP_BASE wxString
     if ( !ch.IsAscii() )
         m_impl = wxStringOperations::EncodeChar(ch);
     else
-#endif // wxUSE_UNICODE_UTF8
+#elif wxUSE_UNICODE_WCHAR && SIZEOF_WCHAR_T == 2
```

Comment:
I think it could be nice to define `wxUSE_UNICODE_UTF16` for this condition, to avoid repeating it all the time and also to make clear what exactly are we testing for here.